seismicpro/src/refractor_velocity/refractor_velocity.py

```diff
@@ -401,34 +370,30 @@ def _calc_init_by_layers(self, n_refractors):
         if n_refractors is None or n_refractors < 1:
             return {}
 
-        max_fb_times = self.fb_times.max()  # times normalization parameter.
-        initial_slope = 4 / 5  # base slope corresponding velocity is 1.25 km/s (v = 1 / slope)
-        initial_slope = initial_slope * self.max_offset / max_fb_times  # add normalization
-        initial_time = 0
-        normalized_offsets = self.offsets / self.max_offset
-        normalized_times = self.fb_times / max_fb_times
-
-        cross_offsets = np.linspace(0, 1, num=n_refractors + 1)  # split cross offsets on an equal intervals
+        cross_offsets = np.linspace(0, self.max_offset, num=n_refractors + 1)
         current_slope = np.empty(n_refractors)
-        current_time = np.empty(n_refractors)
-
-        for i in range(n_refractors):  # inside the loop work with normalized data only
-            mask = (normalized_offsets > cross_offsets[i]) & (normalized_offsets <= cross_offsets[i + 1])
-            if mask.sum() > 1:  # at least two point to fit
-                current_slope[i], current_time[i] = \
-                    self._fit_regressor(normalized_offsets[mask].reshape(-1, 1), normalized_times[mask],
-                                        initial_slope, initial_time)
+        init_t0 = 0
+
+        for i in range(n_refractors):
+            mask = (self.offsets > cross_offsets[i]) & (self.offsets <= cross_offsets[i + 1])
+            # data normalization occurs independently for each layer
+            scaled_offsets, mean_offset, std_offset = self._scale_standard(self.offsets[mask])
+            scaled_times, mean_time, std_time = self._scale_standard(self.fb_times[mask])
+            if std_offset not in [0, np.nan] and std_time not in [0, np.nan]:
```

Comment:
```suggestion
            if std_offset and std_time:
```
And return 0 std in `_scale_standard` for empty data